azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusExtractor.java

```diff
@@ -0,0 +1,527 @@
+  }
+
+  /**
+   * For each of the Conditions present in the the event, extract the corresponding status.
+   * Optional.empty() denotes the corresponding condition was not present.
+   */
+  private void extractConditionStatuses() {
+    this.scheduledCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.scheduledConditionStatus = Optional.of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+    this.containersReadyCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.containersReadyConditionStatus = Optional
+          .of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+    this.initializedCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.initializedConditionStatus =
+          Optional.of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+    this.readyCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.readyConditionStatus = Optional.of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+  }
+
+  /**
+   * Extract the Pod Phase from the event.
+   * @see
+   * <a href="https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/">Pod Lifecycle</a>
+   */
+  private void extractPhase() {
+    requireNonNull(this.v1PodStatus.getPhase(), "pod status phase must not be null");
+    // This will throw an IllegalArgumentException in case of an unexpected phase name.
+    this.podPhase = PodPhase.valueOf(v1PodStatus.getPhase());
+  }
+
+  /**
+   * Evaluate if event can be classified as status {@link AzPodStatus.AZ_POD_REQUESTED}
+   * @return true if the classification succeeds, false otherwise.
+   */
+  private boolean checkForAzPodRequested() {
+    // Scheduled conditions should either not be present or be false
+    if (scheduledConditionStatus.isPresent() &&
+        scheduledConditionStatus.get() == PodConditionStatus.True) {
+      logger.debug("PodRequested is false as scheduled conditions is true");
+      return false;
+    }
+    logger.debug("PodRequested is true");
+    return true;
+  }
+
+  /**
+   * Evaluate if event can be classified as status {@link AzPodStatus.AZ_POD_SCHEDULED}
+   * @return true if the classification succeeds, false otherwise.
+   */
+  private boolean checkForAzPodScheduled() {
+    // Pod must have been scheduled
+    if (!scheduledConditionStatus.isPresent()) {
+      logger.debug("PodScheduled false as scheduled condition is not present");
+      return false;
+    }
+    if (scheduledConditionStatus.get() != PodConditionStatus.True) {
+      logger.debug("PodScheduled false as scheduled condition is not true");
+      return false;
+    }
+    // Initialized condition is not present
+    if (!initializedCondition.isPresent()) {
+      logger.debug("PodScheduled true as initialized condition is not present");
+      return true;
+    }
+    // Initialized condition is not true
```

Comment:
initialized condition is true, right?

Reply:
The comment effectively asserts that 'initialized condition' should be false. Which is what we want in this case, otherwise on the next line if the initialized-condition check return true, the function will return early with the result false.